### DISCUSSION TITLE
Using GPT-4

### DISCUSSION THREAD

**Discussion by wenger9** [sinaptik-ai/pandas-ai#156] [Q&A] [answered]
Hi! Thanks for putting together such a great library.

I am testing pandasai on a google collar notebook. How can I switch between the currently supported models? For example, I'd like to try gpt-4.

Thanks!

**Comment by mspronesti** [comment, marked as answer]
just set the `model` parameter in the `OpenAI` class when instiating it, e.g.
```py
from pandasai import PandasAI
from pandasai.llm import OpenAI

llm = OpenAI(api_key='YOUR_KEY_HERE', model='gpt-4')
pandas_ai = PandasAI(llm, conversational=False)
pandas_ai(df, prompt='YOUR_PROMPT_HERE')
````
if you use `AzureOpenAI` just create a deployment of GPT-4 and set `deployment_name` to the deployment name.

Hope this helps! If it does, please accept my answer 😄

**Reply by wenger9** [reply]
Thank you very much!

**Reply by mspronesti** [reply]
Glad it helped :)
